packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -189,51 +193,67 @@ export const prepareAuctionBook = (baggage, zcf) => {
          * @param {Amount<'nat'>} collateralWanted
          */
         settle(seat, collateralWanted) {
-          const { collateralSeat, curAuctionPrice, currencySeat } = this.state;
-          const { Currency: currencyAvailable } = seat.getCurrentAllocation();
+          const { collateralSeat, collateralBrand } = this.state;
+          const { Currency: currencyAlloc } = seat.getCurrentAllocation();
           const { Collateral: collateralAvailable } =
             collateralSeat.getCurrentAllocation();
           if (!collateralAvailable || AmountMath.isEmpty(collateralAvailable)) {
-            return AmountMath.makeEmptyFromAmount(collateralWanted);
+            return makeEmpty(collateralBrand);
           }
 
           /** @type {Amount<'nat'>} */
-          const collateralTarget = AmountMath.min(
+          const initialCollateralTarget = AmountMath.min(
             collateralWanted,
             collateralAvailable,
           );
 
+          const { curAuctionPrice, currencySeat, totalProceedsGoal } =
+            this.state;
           const currencyNeeded = ceilMultiplyBy(
-            collateralTarget,
+            initialCollateralTarget,
             curAuctionPrice,
           );
           if (AmountMath.isEmpty(currencyNeeded)) {
-            seat.fail('price fell to zero');
-            return AmountMath.makeEmptyFromAmount(collateralWanted);
+            seat.fail(Error('price fell to zero'));
+            return makeEmpty(collateralBrand);
           }
 
-          const affordableAmounts = () => {
-            if (AmountMath.isGTE(currencyAvailable, currencyNeeded)) {
-              return [collateralTarget, currencyNeeded];
-            } else {
-              const affordableCollateral = floorDivideBy(
-                currencyAvailable,
-                curAuctionPrice,
-              );
-              return [affordableCollateral, currencyAvailable];
-            }
-          };
-          const [collateralAmount, currencyAmount] = affordableAmounts();
-          trace('settle', { collateralAmount, currencyAmount });
+          const initialCurrencyTarget = AmountMath.min(
+            currencyNeeded,
+            currencyAlloc,
+          );
+          const currencyLimit = totalProceedsGoal
+            ? AmountMath.min(totalProceedsGoal, initialCurrencyTarget)
+            : initialCurrencyTarget;
+          const isRaiseBounded =
```

Comment:
"bounded" might not be worth a new term.
```suggestion
          const isRaiseLimited =
```